pkg/agent/route/route_windows.go

```diff
@@ -387,16 +513,80 @@ func (c *Client) DeleteSNATRule(mark uint32) error {
 
 // TODO: nodePortAddresses is not supported currently.
 func (c *Client) AddNodePort(nodePortAddresses []net.IP, port uint16, protocol binding.Protocol) error {
-	return util.ReplaceNetNatStaticMapping(antreaNatNodePort, "0.0.0.0", port, config.VirtualServiceIPv4.String(), port, string(protocol))
+	netNatStaticMapping := &util.NetNatStaticMapping{
+		Name:         antreaNatNodePort,
+		ExternalIP:   net.ParseIP("0.0.0.0"),
+		ExternalPort: port,
+		InternalIP:   config.VirtualNodePortDNATIPv4,
+		InternalPort: port,
+		Protocol:     protocol,
+	}
+	if err := util.ReplaceNetNatStaticMapping(netNatStaticMapping); err != nil {
+		return err
+	}
+	c.netNat.Store(fmt.Sprintf("%d-%s", port, protocol), netNatStaticMapping)
+	klog.V(4).InfoS("Added NodePort NetNatStaticMapping", "NetNatStaticMapping", netNatStaticMapping)
+	return nil
 }
 
 func (c *Client) DeleteNodePort(nodePortAddresses []net.IP, port uint16, protocol binding.Protocol) error {
-	return util.RemoveNetNatStaticMapping(antreaNatNodePort, "0.0.0.0", port, string(protocol))
+	obj, found := c.netNat.Load(fmt.Sprintf("%d-%s", port, protocol))
+	if !found {
+		klog.V(2).Infof("NetNatStaticMapping with port %d, protocol %s not exists", port, protocol)
+		return nil
+	}
+	netNatStaticMapping := obj.(*util.NetNatStaticMapping)
+	if err := util.RemoveNetNatStaticMapping(netNatStaticMapping); err != nil {
+		return err
+	}
+	c.netNat.Delete(strconv.Itoa(int(port)))
```

Comment:
The key in `netNat` is `fmt.Sprintf("%d-%s", port, protocol)`, but here you use port number?

Reply:
Thanks, updated.